### PR TITLE
refactor(samples): Simplify samples

### PR DESCRIPTION
**Pull request by timtay-microsoft**
- Stop using async APIs followed by a Thread.sleep, just use sync APIs
- Make device/module client open calls use retry
- Remove random Thread.sleep calls
- Better demonstrate catching IothubClientException, getting status code
- Remove "send serialized event" sample because it didn't demonstrate anything meaningful about this SDK

### CODE REVIEW

**Review by patilsnr — APPROVED** [review]

File: iothub/device/iot-device-samples/send-receive-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SendReceive.java [hunk 1]
```diff
@@ -21,114 +21,21 @@
...
-        }
-
-        @Override
-        public String toString()
-        {
-            return Integer.toString(this.num);
-        }
-    }
-
-    protected static class MessageCallback
```
[the remark this answers is not in the document]
Comment (timtay-microsoft) [reply iothub/device/iot-device-samples/send-receive-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SendReceive.java:53]: It's mostly that it is difficult to read a sample unless you inline these kinds of declarations. A user would see this class, not see it being used until much farther down in the sample, then they'd likely scroll back up just to see what this class is. It makes more sense for a sample to declare this callback implementation exactly where it is being used

File: iothub/device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/TemperatureController.java [hunk 2]
```diff
@@ -250,41 +229,52 @@ private static void initializeAndProvisionDevice() throws ProvisioningDeviceClie
...
+                registrationResultReference.set(provisioningDeviceClientRegistrationResult);
+                registrationExceptionReference.set(e);
+                registrationLatch.countDown();
+            },
+            null,
+            additionalData);
 
-        while (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() != ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ASSIGNED)
+        System.out.println("Waiting for Provisioning Service to register");
+        boolean timedOut = !registrationLatch.await(1, TimeUnit.MINUTES);
```
Comment (andyk-ms) [inline iothub/device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/TemperatureController.java:252]: Is this timeout a best practice approach?
Is there any documentation or additional comments we can provide about the choice of this 'timeout'?

File: iothub/device/iot-device-samples/send-event-with-proxy/src/main/java/samples/com/microsoft/azure/sdk/iot/SendEventWithProxy.java [hunk 3]
```diff
@@ -18,21 +18,7 @@
...
-            IotHubStatusCode status = exception == null ? IotHubStatusCode.OK : exception.getStatusCode();
-            System.out.println("IoT Hub responded to message "+ sentMessage.getMessageId()  + " with status " + status.name());
-
-            if (status==IotHubStatusCode.MESSAGE_CANCELLED_ONCLOSE)
-            {
-                failedMessageListOnClose.add(sentMessage.getMessageId());
-            }
-        }
-    }
+    private  static final int D2C_MESSAGE_TIMEOUT = 2000; // 2 seconds
```
Comment (andyk-ms) [inline iothub/device/iot-device-samples/send-event-with-proxy/src/main/java/samples/com/microsoft/azure/sdk/iot/SendEventWithProxy.java:21]: additional comment about the choice of this timeout will be very helpful.

File: iothub/device/iot-device-samples/send-receive-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SendReceive.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (patilsnr) [reply iothub/device/iot-device-samples/send-receive-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SendReceive.java:53]: Makes sense, I've felt similar in the past. As long as we're not getting rid of anything that could somehow break the sample, I'm all for it.

**Pull request merged**